Reconfigure .pro to install the .so and headers on unix/mac

EvanPurkhiser opened the pull request:
In relation to [my comment here](https://github.com/ibsh/libKeyFinder/commit/571628e7eff6bab4e8434b28470f87c6ce369b8f#commitcomment-13177855)

This fixes the project file so that it correctly generates an install target on OSX and Unix for both the library file and the header files.

Platform | Library Path | Headers Path
---|---|---
**OSX** | `/usr/local/lib` | `/usr/local/include/keyfinder`
**Unix** | `/usr/lib` | `/usr/include/keyfinder`

I can't say that I'm 100% sure on what the `QMAKE_LFLAGS_SONAME` definition does, but I do know that using it on unix causes the linker to fail. I left it in for OSX, but I'm *pretty sure* OSX wasn't installing the library files either since it wasn't included in the `INSTALL`. Full disclosure though, I didn't fully test this on OSX since I'm having trouble getting qt working :frowning:

ibsh commented:
Cool, I'll have to take a look on the Mac to check this one, which will probably take some time; crazy busy right now. But thanks for submitting.

EvanPurkhiser commented:
Thanks, not a problem :-)

fitzgen commented:
I think I ran into this issue as well. I had to manually symlink the dyilb file into `/usr/local/lib` on OSX.

Will try out this patch and confirm the fix.

fitzgen commented:
Awesome! The patch fixed the install problems I was seeing!

Less awesome: when I run the tests I get a failure that did not exist when I didn't have this patch applied:

    $ ./tests    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    tests is a Catch v1.0 b53 host application.
    Run with -? for options

    -------------------------------------------------------------------------------
    FftAdapterTest/ForwardAndBackward
    -------------------------------------------------------------------------------
    fftadaptertest.cpp:24
    ...............................................................................

    fftadaptertest.cpp:59: FAILED:
      REQUIRE( (5) > (out) )
    with expansion:
      5 > 896.34735f

    ===============================================================================
    test cases:      65 |      64 passed | 1 failed
    assertions: 1695272 | 1695271 passed | 1 failed

fitzgen commented:
Yeah, just confirmed again that if I don't apply this patch and instead symlink the dylib from `/path/to/libKeyFinder/libkeyfinder.dylib` to `/usr/local/lib/libkeyfinder.dylib` I do not get the test failure:

    ./tests 
    ===============================================================================
    All tests passed (1705510 assertions in 65 test cases)

ibsh commented:
Interesting, guys. I've found that test failure to be intermittent in my own work but wasn't able to tie it down to a specific cause... if you can figure it out please let me know. I don't have any time to work on KF at the moment and appreciate your investment.

fitzgen commented:
Thanks for writing and sharing the library!

I ran the tests twice and it failed both times, maybe I was just unlucky.

Are you still able to make time for reviewing pull requests? If not, are there any consistent contributors you could add as a committer to handle PRs so that the library can continue to evolve?

Thanks again!

ibsh commented:
I can usually review pulls depending on their complexity and my familiarity with their subject matter. Your [#10](https://github.com/ibsh/libKeyFinder/pull/10) for example is super easy. This one requires a bit too much research given Other Stuff Happening.

ibsh commented:
Oh, and as for the test failing intermittently or consistently; obviously I'd prefer for it not to fail at all! It is a test which, from a purely logical perspective, should never fail, so it's more testing the build process at the moment...

fitzgen commented:
Ok, so I just tried once more and found that the test fails intermittently both with my symlink hack and with this PR merged.

I would recommend disregarding my earlier comments and merging this pull request.

EvanPurkhiser commented:
Probably good this hasn't been merged yet.

I seem to have discovered an issue in the configuration of the `target.path` and `header.path` values. These were configured for the `unix` target *after* the `macx` target, overwriting the values. So doing a `make install` on OS X attempted to install things right into `/usr`.

I've updated the pull request to correct the target that changes the paths for Linux to *not* match OS X.

I've also rebased onto of master.

Hopefully this can be verified and merged soon :-)

iammordaty commented:
Are there any chances formie merge?

iammordaty commented:
Thanks!
